src/item.cpp

```diff
@@ -6921,6 +6921,16 @@ bool item::is_reloadable_helper( const itype_id &ammo, bool now ) const
         return true;
     }
 
+    if( is_watertight_container() && !contents.empty() &&
+        ( contents.only_item().typeId() == ammo ) ) {
```

Comment:
This will show a debug message if there's more than one item in the container. You should check the stack size is 1 before calling `only_item`.

Reply:
It would be so much easier to be sure that only one item stack is provided. If I were to put a test condition for only one stack that still leaves the problem of what to do with stack size greater than 1. 

The method ::legacy_front() does not look so bad any more.

Reply:
A watertight container can contain more than one solid item stack in normal gameplay if you manually insert items into the container. In that case `only_item` and `legacy_front` will both show a debug message complaining there are more than one item stack in the container.

>  what to do with stack size greater than 1

My suggestion is to disable reloading if a container contains non-liquid or more than one item stack (perhaps in `item::is_reloadable`).

Reply:
please don't do new code with legacy functions, they're that way in order to eventually be phased out when possible.

Reply:
Disabling of reload when there is more than one stack is the most sensible way to approach this. 

Thanks for the feedback; it is invaluable.